Python/Python.sublime-syntax

```diff
@@ -364,162 +546,287 @@ contexts:
+  function-parameter-tuples:
+    # python 2 style tuple arguments
+    # removed from python 3 since PEP-3113
+    - match: \(
+      scope: punctuation.section.group.begin.python
+      push: function-parameter-tuple-body
+
+  function-parameter-tuple-body:
+    - meta_scope: meta.group.python
+    - match: \)
+      scope: punctuation.section.group.end.python
+      set: after-expression
+    - match: ','
+      scope: punctuation.separator.parameters.python
+      push: allow-unpack-operators
+    # default values should follow the argument
+    - match: '{{assignment_operator}}'
+      push: function-parameter-illegal-annotation
+    # python 2 does not support type annotations
+    - match: '{{colon}}'
+      push: function-parameter-illegal-default-value
+    - include: parameter-names
+    - include: line-continuations
+    - include: illegal-assignment-expressions
+
+  function-parameter-illegal-annotation:
+    - meta_include_prototype: false
+    - meta_scope: invalid.illegal.default-value.python
+    - match: (?=[,)=])
+      pop: 1
+
+  function-parameter-illegal-default-value:
+    - meta_include_prototype: false
+    - meta_scope: invalid.illegal.annotation.python
+    - match: (?=[,)=])
+      pop: 1
+
+  function-after-parameters:
+    - meta_content_scope: meta.function.python
+    - match: ->
+      scope: meta.function.annotation.return.python punctuation.separator.annotation.return.python
+      set: function-return-type
+    - include: function-definition-end
+
+  function-return-type:
+    - meta_content_scope: meta.function.annotation.return.python
+    - include: function-definition-end
+    - include: expression-in-a-statement
+
+  function-definition-end:
+    - match: '{{colon}}'
+      scope: meta.function.python punctuation.section.function.begin.python
+      pop: 1
+    - include: illegal-assignment-expressions
+    - include: line-continuation-or-pop
+
+###[ ASSIGNMENT STATEMENTS ]##################################################
+
+  assignment-statements:
+    - match: '{{colon}}'
+      scope: punctuation.separator.annotation.variable.python
+      push: variable-annotation
+    - match: '{{augmented_assignment_operators}}'
+      scope: keyword.operator.assignment.augmented.python
+      push: assignment-statement-value
+    - match: '{{assignment_operator}}'
+      scope: keyword.operator.assignment.python
+      push: assignment-statement-value
+
+  assignment-statement-value:
+    - include: line-continuation-or-pop
+    - include: illegal-assignment-expressions
+    - include: expression-in-a-statement
+
+  variable-annotation:
+    - meta_scope: meta.variable.annotation.python
+    - match: (?=$|=|#|;)
+      pop: 1
+    - match: None\b
+      scope: constant.language.null.python
+    - include: expression-in-a-group
+
+###[ CASE STATEMENTS ]########################################################
+
+  case-statements:
+    # https://www.python.org/dev/peps/pep-0634
+    - match: (?=case\b)
```

Comment:
In which way is it inconsistent? This is the way it needs to be handled in branch points in case a keyword is to be scoped differently. See how all the group vs. generator or dict vs. set branch points are implemented.

Also this implementation hasn't changed compared to `master`.